data/expenses.js

```diff
@@ -0,0 +1,152 @@
+const { ObjectId, ObjectID } = require('mongodb');
+const { expenses: getExpensesCollection } = require('../config/mongoCollections');
+
+const { QueryError, ValidationError } = require('../utils/errors');
+const { idQuery, parseMongoData, stringifyObjectId } = require('../utils/mongodb');
+const {
+  assertObjectIdString,
+  assertIsValuedString,
+  assertRequiredObject,
+} = require('../utils/assertion');
+
+const getByObjectId = async (objectId) => {
+  const collection = await getExpensesCollection();
+  const expense = await collection.findOne(idQuery(objectId));
+  return parseMongoData(expense);
+};
+
+const getExpense = async (expenseId) => {
+  assertObjectIdString(expenseId, 'Expense id');
+
+  let expense = await getByObjectId(expenseId);
+  if (expense == null) {
+    throw new QueryError(`Could not get expense for (${expenseId})`);
+  }
+  return expense;
+};
+
+const getAllExpenses = async (tripId) => {
```

Comment:
I think the naming should reflect this data function's purpose. There could be get all expenses for this user, or all expenses for this corporate

Reply:
I have changed this, as it is by trip. Shall I do it for all filters like by user, by corporate, and as well as by trip?

Reply:
I think as long as the naming won't cause potential confusing for imports an invokes, it should be fine.